### DISCUSSION TITLE
Best way to get foreign table's count?

### DISCUSSION THREAD

**Discussion by imtaehyun** [supabase/supabase#669] [Questions] [answered]
Hi,

Thanks for recent update (getting count feature) of library. It's super very cool.

I have a question regarding this feature.
What is the best way to get foreign table's count?

I tried with the codes below.
```javascript
// tables: notes, comments
// comments table has a foreign key referencing notes table
let { data: notes, error } = await supabase
          .from("notes")
          .select(
            `
          id,
          title,
          description,
          createdAt,
          comments(
            count
          )`
          )

// note is one element of notes array
const commentsCount = note.comments[0].count
```

To access `count`field, it need to access `comments` array first. I'm just curious is there efficient way to get `count` of foreign table.

Thanks

**Comment by steve-chavez** [comment, marked as answer]
Hey @imtaehyun,

For now it's only possible to get the count through the array.

Later on it'll be possible to "flatten" the embedded resource so you can get the count like:

```js
let { data: notes, error } = await supabase
          .from("notes")
          .select(
            `
          id,
          title,
          description,
          createdAt,
          commentsCount:comments(count)->count`
          )

const commentsCount = note.commentsCount
```

Pending enhancement tracked on: https://github.com/PostgREST/postgrest/issues/1233#issuecomment-469782724
